sys/kern/makesyscalls.sh

```diff
@@ -956,11 +956,12 @@ sed -e '
 					comma = ""
 				else
 					comma = ", "
-				if (isptrtype(argtype[i]) && !(argtype[i] ~ /caddr_t/)) {
+				if (isptrtype(argtype[i]) && !(argtype[i] ~ /caddr_t/) && !(argtype[i] ~ /intptr_t/)) {
 					a_type = argtype[i]
 					sub(/_c /, "", a_type)
 					sub(/_c_/, "_", a_type)
-					cast = "(" a_type ")"
+					# THIS IS NOT CORRECT, only works for cheriabi!
```

Comment:
Looking at `lib/libcheri/cheri_system.c` this looks right if I understand `__cheri_cast` correctly.  When `libcheri` is hybrid this will correctly cast away the capability.  When it's cheriabi, it will be a noop which is correct to make a syscall.  It's ok that this is cheri specific as the syscall stuff stuff is inherently cheri specific.

Reply:
Okay I wasn't sure whether this code would also be used to generate other files other than the cheriabi stubs.